common/djangoapps/third_party_auth/models.py

```diff
@@ -226,7 +237,27 @@ def display_for_login(self):
         Determines whether the provider ought to be shown as an option with
         which to authenticate on the login screen, registration screen, and elsewhere.
         """
-        return bool(self.enabled and self.accepts_logins and self.visible)
+        return bool(self.usable and self.accepts_logins and self.visible)
+
+    @property
+    def for_current_site(self):
+        """
+        Determines if the provider is configured to be used with the current site.
+        """
+        if configuration_helpers.is_site_configuration_enabled():
+            # If there's a site enabled for this request, then return True if the site matches.
+            config = configuration_helpers.get_current_site_configuration()
+            return config.site == self.site
+        # If not viewing a site, return True if the site is unset (which signifies that the provider is intended for
+        # the default non-site platform).
+        return self.site is None
+
+    @property
+    def usable(self):
```

Comment:
Yes, I've combined them. I initially wrote them separately as I anticipated needing to use both of them, but it turns out that `usable` is the only check necessary